src/Router.php

```diff
@@ -326,6 +326,10 @@ protected static function get_response_headers() {
 			]
 		);
 
+		// For cache url header, use the domain without protocol or path.
+		$graphql_endpoint  = wp_parse_url( graphql_get_endpoint_url() );
+		$endpoint_hostname = isset( $graphql_endpoint['host'] ) ? $graphql_endpoint['host'] : graphql_get_endpoint_url();
```

Comment:
`graphql_get_endpoint_url()` applies a filter, so we should avoid calling it twice if we can.

```suggestion
		$graphql_endpoint  = graphql_get_endpoint_url() ;
		$endpoint_hostname = wp_parse_url( $graphql_endpoint, PHP_URL_HOST ) ?: $graphql_endpoint;
```